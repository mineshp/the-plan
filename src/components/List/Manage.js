import React from 'react';
import { Container } from 'semantic-ui-react';
import ManageList from '../../HOC/List/ManageList';

const Manage = () => (
    <Container>
        <ManageList />
    </Container>
);

export default Manage;
