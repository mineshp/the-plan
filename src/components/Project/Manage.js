import React from 'react';
import ManageProject from '../../HOC/Project/ManageProject.js';
import LeftHandMenu from '../Shared/LeftHandMenu.js';
import { Container, Grid } from 'semantic-ui-react';

const Manage = () => (
  <Container>
    <Grid>
      <Grid.Column width={4}>
        <LeftHandMenu />
      </Grid.Column>
      <Grid.Column width={12}>
        <ManageProject />
      </Grid.Column>
    </Grid>
  </Container>
)

export default Manage;