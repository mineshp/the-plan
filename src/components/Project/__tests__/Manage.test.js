import React from 'react';
import Manage from '../Manage';
import renderer from 'react-test-renderer';

const mockListAllProjects = [
    {
        _id: "5992d50066a7043f2598e12d",
        projectName: "PROJECT 1",
        colour: "red",
        createdDate: "2017-08-15T12:02:00.000Z"
    },
    {
        _id: "59a08710e57cb1da97cd1477",
        projectName: "PROJECT 2",
        colour: "red",
        __v: 0,
        createdDate: "2017-08-25T20:22:40.994Z"
    },
    {
        _id: "59a08bb6e57cb1da97cd1478",
        projectName: "PROJECT 3",
        colour: "grey",
        __v: 0,
        createdDate: "2017-08-25T20:42:30.159Z"
    }
];

const mockResponse = (status, statusText, response) => {
  return new window.Response(response, {
    status: status,
    statusText: statusText,
    headers: {
      'Content-type': 'application/json'
    }
  });
};

describe('Manage Project', () => {
    beforeEach(() => {
        window.fetch = jest.fn().mockImplementation(() =>
            Promise.resolve(mockResponse(
                200, null, JSON.stringify(mockListAllProjects) )));
    });

    it('renders correctly', () => {
        const tree = renderer.create(
            <Manage></Manage>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
