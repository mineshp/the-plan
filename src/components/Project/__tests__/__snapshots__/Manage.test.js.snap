// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Manage Project renders correctly 1`] = `
<div
  className="ui container"
>
  <div
    className="ui grid"
  >
    <div
      className="four wide column"
    >
      <div
        className="ui vertical labeled icon menu"
      >
        <a
          className="active item"
          onClick={[Function]}
        >
          <i
            aria-hidden="true"
            className="list layout icon"
          />
          View
        </a>
        <a
          className="item"
          href="/project/create"
          onClick={[Function]}
        >
          <i
            aria-hidden="true"
            className="add circle icon"
          />
          Create
        </a>
        <a
          className="item"
          onClick={[Function]}
        >
          <i
            aria-hidden="true"
            className="search icon"
          />
          Search
        </a>
      </div>
    </div>
    <div
      className="twelve wide column"
    >
      <div
        className="List main"
      >
        <div
          className="ui three cards"
        />
      </div>
    </div>
  </div>
</div>
`;
